### PR TITLE
Deprecate dao methods that do not take a RequestDetails parameter

### PR DESCRIPTION
**Pull request by fil512**

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java [hunk 1]
```diff
@@ -200,6 +203,9 @@ public DaoMethodOutcome create(final T theResource, RequestDetails theRequestDet
 		return create(theResource, null, true, new TransactionDetails(), theRequestDetails);
 	}
 
+	/**
+	 * @deprecated Use {@link #create(T, String, RequestDetails)} instead
```
Comment (jmarchionatto) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java:207]: question: what if theIfNotExists parameter is needed?

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java [hunk 2]
```diff
@@ -1287,12 +1302,6 @@ public BaseHasResource readEntity(IIdType theId, RequestDetails theRequest) {
 		return readEntity(theId, true, theRequest);
 	}
 
-	@Override
```
Comment (jmarchionatto) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java:1290]: question: can't see why this override was removed

**Review by jmarchionatto — APPROVED** [review]
Approved pending a couple of questions

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java (hunk 2 above)
Comment (fil512) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java:1290]: The method is an implementation of an unused method in the interface.  I deprecated the interface method.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java (hunk 1 above)
Comment (fil512) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java:207]: That's what the second `String` parameter is.

**Pull request merged**